*** acts_as_searchable.rb.orig	2008-06-22 01:08:36.000000000 +0900
--- acts_as_searchable.rb	2008-06-22 01:17:43.000000000 +0900
***************
*** 127,145 ****
            self.attributes_to_store  = options[:attributes] || {}
            self.if_changed           = options[:if_changed] || []
            
-           send :attr_accessor, :changed_attributes
- 
            class_eval do
              after_update  :update_index
              after_create  :add_to_index
              after_destroy :remove_from_index
-             after_save    :clear_changed_attributes
- 
-             (if_changed + searchable_fields + attributes_to_store.collect { |attribute, method| method or attribute }).each do |attr_name|
-               define_method("#{attr_name}=") do |value|
-                 write_changed_attribute attr_name, value
-               end
-             end
  
              connect_estraier
            end
--- 127,136 ----
***************
*** 290,313 ****
            get_doc_from(result)
          end
          
-         # If called with no parameters, gets whether the current model has changed and needs to updated in the index.
-         # If called with a single parameter, gets whether the parameter has changed.
-         def changed?(attr_name = nil)
-           changed_attributes and (attr_name.nil? ?
-             (not changed_attributes.length.zero?) : (changed_attributes.include?(attr_name.to_s)) )
-         end
-         
          protected
          
-         def clear_changed_attributes #:nodoc:
-           self.changed_attributes = []
-         end
-         
-         def write_changed_attribute(attr_name, attr_value) #:nodoc:
-           (self.changed_attributes ||= []) << attr_name.to_s unless self.changed?(attr_name) or self.send(attr_name) == attr_value
-           write_attribute(attr_name.to_s, attr_value)
-         end
- 
          def add_to_index #:nodoc:
            seconds = Benchmark.realtime { estraier_connection.put_doc(document_object) }
            logger.debug "#{self.class.to_s} [##{id}] Adding to index (#{sprintf("%f", seconds)})"
--- 281,288 ----
